lambda/controller/create-replica/index.js

```diff
@@ -24,26 +24,42 @@ exports.handler = function(event, context, callback){
       return callback(err);
     }
 
+    var billingMode = 'PROVISIONED';
+
+    // Check the billing mode of the table
+    if (data.Table.hasOwnProperty('BillingModeSummary')){
+      if (data.Table.BillingModeSummary.BillingMode == 'PAY_PER_REQUEST') {
+        // console.log('PAY_PER_REQUEST');
+        billingMode = 'PAY_PER_REQUEST';
+      }
+    }
     //Construct replica table using source table description
     var params = {
       TableName: data.Table.TableName,
       AttributeDefinitions: data.Table.AttributeDefinitions,
       KeySchema: data.Table.KeySchema,
-      ProvisionedThroughput: {
-        ReadCapacityUnits: data.Table.ProvisionedThroughput.ReadCapacityUnits,
-        WriteCapacityUnits: data.Table.ProvisionedThroughput.WriteCapacityUnits
-      },
       StreamSpecification: {
         StreamEnabled: true,
         StreamViewType: 'NEW_AND_OLD_IMAGES'
       }
     };
 
+    // if the billingMode is PROVISIONED, need to add the throughput numbers
+    if (billingMode == 'PROVISIONED') {
+      params.ProvisionedThroughput = {
+        ReadCapacityUnits: data.Table.ProvisionedThroughput.ReadCapacityUnits,
+        WriteCapacityUnits: data.Table.ProvisionedThroughput.WriteCapacityUnits
+      };
+    } else {
+      // otherwise set the billingMode to PAY_PER_REQUEST
+      params.BillingMode = 'PAY_PER_REQUEST'
```

Comment:
Minor: In theory your billingMode variable should only be `PROVISIONED` or `PAY_PER_REQUEST` both of which are accepted parameters in the createTable SDK call (https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/DynamoDB.html#createTable-property). As such you could just always include `BillingMode` in the initial params block and remove this else block entirely. 

i.e. 
```
var params = {
      TableName: data.Table.TableName,
      AttributeDefinitions: data.Table.AttributeDefinitions,
      KeySchema: data.Table.KeySchema,
      ProvisionedThroughput: {
        ReadCapacityUnits: data.Table.ProvisionedThroughput.ReadCapacityUnits,
        WriteCapacityUnits: data.Table.ProvisionedThroughput.WriteCapacityUnits
      },
      StreamSpecification: {
        StreamEnabled: true,
        StreamViewType: 'NEW_AND_OLD_IMAGES'
      }
    };
```

becomes 

```
var params = {
      TableName: data.Table.TableName,
      BillingMode: billingMode, 
      AttributeDefinitions: data.Table.AttributeDefinitions,
      KeySchema: data.Table.KeySchema,
      ProvisionedThroughput: {
        ReadCapacityUnits: data.Table.ProvisionedThroughput.ReadCapacityUnits,
        WriteCapacityUnits: data.Table.ProvisionedThroughput.WriteCapacityUnits
      },
      StreamSpecification: {
        StreamEnabled: true,
        StreamViewType: 'NEW_AND_OLD_IMAGES'
      }
    };
```

Reply:
Hmmm...is it a required parameter though? In my testing, I didn't need to provide it in the PROVISIONED case. Your point is well taken though - no harm in adding it. I'll test first.